impacted-test-engine/src/test/java/com/teamscale/test_impacted/engine/ImpactedTestEngineWithTwoEnginesTest.java

```diff
@@ -0,0 +1,152 @@
+package com.teamscale.test_impacted.engine;
+
+import com.teamscale.client.PrioritizableTest;
+import com.teamscale.client.PrioritizableTestCluster;
+import com.teamscale.test_impacted.engine.executor.DummyEngine;
+import com.teamscale.test_impacted.test_descriptor.JUnitJupiterTestDescriptorResolver;
+import org.junit.jupiter.api.Disabled;
+import org.junit.platform.engine.EngineExecutionListener;
+import org.junit.platform.engine.TestDescriptor;
+import org.junit.platform.engine.TestEngine;
+import org.junit.platform.engine.TestExecutionResult;
+import org.junit.platform.engine.UniqueId;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static com.teamscale.test_impacted.engine.executor.SimpleTestDescriptor.testCase;
+import static com.teamscale.test_impacted.engine.executor.SimpleTestDescriptor.testContainer;
+import static java.util.Arrays.asList;
+import static java.util.Collections.singletonList;
+import static org.junit.platform.engine.TestExecutionResult.failed;
+import static org.junit.platform.engine.TestExecutionResult.successful;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.verify;
+
+/** Test setup for a mixture of impacted and no impacted tests and two test engines. */
+class ImpactedTestEngineWithTwoEnginesTest extends ImpactedTestEngineTestBase {
+
+	/**
+	 * For this test setup we rely on the {@link JUnitJupiterTestDescriptorResolver} for resolving uniform paths and
+	 * cluster ids. Therefore, the engine root is set accordingly.
+	 */
+	private final UniqueId engine1RootId = UniqueId.forEngine("junit-jupiter");
+	private final UniqueId engine2RootId = UniqueId.forEngine("exotic-engine");
+
+	/** FirstTestClassDiscovery contains a one impacted and one non-impacted test. */
+	private final UniqueId firstTestClassId = engine1RootId.append("class", "FirstTestClass");
```

Comment:
No there is no constant and even if there was we can't use them, because we don't have a direct dependency on the Jupiter Engine. I centralized all segment types now in `JUnitJupiterTestDescriptorResolver` and `JUnitVintageTestDescriptorResolver`.